src/widgets/maybe.rs

```diff
@@ -0,0 +1,167 @@
+        none_maker: impl Fn() -> W2 + 'static,
+    ) -> Maybe<T>
+    where
+        W1: Widget<T> + 'static,
+        W2: Widget<()> + 'static,
+    {
+        let widget = MaybeWidget::Some(WidgetPod::new(some_maker().boxed()));
+        Maybe {
+            some_maker: Box::new(move || some_maker().boxed()),
+            none_maker: Box::new(move || none_maker().boxed()),
+            widget,
+        }
+    }
+
+    /// Create a new `Maybe` widget where the `None` branch is an empty widget.
+    #[allow(dead_code)]
+    pub fn or_empty<W1: Widget<T> + 'static>(some_maker: impl Fn() -> W1 + 'static) -> Maybe<T> {
+        Self::new(some_maker, || SizedBox::empty())
+    }
+
+    fn rebuild_widget(&mut self, is_some: bool) {
+        if is_some {
+            self.widget = MaybeWidget::Some(WidgetPod::new((self.some_maker)()));
+        } else {
+            self.widget = MaybeWidget::None(WidgetPod::new((self.none_maker)()));
+        }
+    }
+}
+
+impl<T: Data> Widget<Option<T>> for Maybe<T> {
+    fn event(&mut self, ctx: &mut EventCtx, event: &Event, data: &mut Option<T>, env: &Env) {
+        match data.as_mut() {
+            Some(d) => self.widget.unwrap_some().event(ctx, event, d, env),
+            None => self.widget.unwrap_none().event(ctx, event, &mut (), env),
+        }
+    }
+
+    fn lifecycle(
+        &mut self,
+        ctx: &mut LifeCycleCtx,
+        event: &LifeCycle,
+        data: &Option<T>,
+        env: &Env,
+    ) {
+        if let LifeCycle::WidgetAdded = event {
+            if data.is_some() != self.widget.is_some() {
+                // only possible at launch, because we default to `Some`
+                self.rebuild_widget(data.is_some());
+            }
+        }
+        match data.as_ref() {
+            Some(d) => self.widget.unwrap_some().lifecycle(ctx, event, d, env),
+            None => self
+                .widget
+                .unwrap_none()
+                .lifecycle(ctx, event, &mut (), env),
+        }
+    }
+
+    fn update(&mut self, ctx: &mut UpdateCtx, old_data: &Option<T>, data: &Option<T>, env: &Env) {
+        if old_data.is_some() != data.is_some() {
+            self.rebuild_widget(data.is_some());
+            ctx.children_changed();
+            ctx.request_layout();
+        } else {
+            match data {
+                Some(new) => self.widget.unwrap_some().update(ctx, new, env),
+                None => self.widget.unwrap_none().update(ctx, &(), env),
+            }
+        }
+    }
+
+    fn layout(
+        &mut self,
+        ctx: &mut LayoutCtx,
+        bc: &BoxConstraints,
+        data: &Option<T>,
+        env: &Env,
+    ) -> Size {
+        match data.as_ref() {
+            Some(d) => {
+                let size = self.widget.unwrap_some().layout(ctx, bc, d, env);
+                self.widget.unwrap_some().set_layout_rect(size.to_rect());
+                ctx.set_paint_insets(self.widget.unwrap_some().paint_insets());
```

Comment:
According to the docs it seems like `set_paint_insets` is not necessary in this case. Curious why you added it.

Reply:
you're right; for some reason I thought that paint insets were cleared at each widgetpod. :)